loki/batch/sgraph.py

```diff
@@ -292,6 +312,36 @@ def dependencies(self):
         """
         return tuple(self._graph.edges)
 
+    @staticmethod
+    def _get_item_filter(item_filter):
+        """
+        If :any:`ProcedureItem` is part of ``item_filter``, add :any:`ProcedureBindingItem` and
+        :any:`InterfaceItem` as well, since these are intermediate nodes. Their
+        dependencies will also be included until they eventually resolve to a
+        :any:`ProcedureItem`.
+
+        This returns the updated ``item_filter``.
+
+        Parameters
+        ----------
+        item_filter : list of :any:`Item` subclasses, optional
+            Filter successor items to only include items of the provided type. By default,
+            all items are considered. Note that including :any:`ProcedureItem` in the
+            ``item_filter`` automatically adds :any:`ProcedureBindingItem` and
+            :any:`InterfaceItem` as well, since these are intermediate nodes. Their
+            dependencies will also be included until they eventually resolve to a
+            :any:`ProcedureItem`.
+        """
+        if ProcedureItem in as_tuple(item_filter):
```

Comment:
```suggestion
        item_filter = as_tuple(item_filter)
        if ProcedureItem in item_filter:
```
Otherwise the conditional in the next line fails if `item_filter` is not a tuple.